diff --git a/HTML/Header.html b/HTML/Header.html
index 1e87c87..a5dca1f 100644
--- a/HTML/Header.html
+++ b/HTML/Header.html
@@ -1,32 +1,32 @@
-<!DOCTYPE html>
-<html lang="en">
-<head>
-    <link rel="stylesheet" type="text/css" media="screen" href="../CSS/ResetSCSS.css" >
-    <link rel="stylesheet" type="text/css" media="screen" href="../CSS/HeaderCSS.css" >
-    <script src = "../JavaScipt/HeaderJS.js"></script>
-    
-    <meta charset="UTF-8">
-    <title>Practica 3</title>
-    <meta name="description" content="Practica 3" />
-    
-</head>
-
-<body>
-    <header class="full" id="header">
-        <span id="x"></span>  
-        
-        <img id = "logo" src="../images/logo.png" width="120" height="40" alt="logo">
-        
-        <img id="lupa" src="../images/busqueda.svg" width="40" height="40" alt="lupa">
-        
-    </header>
-    <main>
-        <div id="menu2" >
-            <p id = "m1"></p>
-            <p id = "m2"></p>
-            <p id = "m3"></p>
-            <p id = "m4"></p>
-            <p id = "m5"></p>
-        </div>
-    </main>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <link rel="stylesheet" type="text/css" media="screen" href="../CSS/ResetSCSS.css" >
+    <link rel="stylesheet" type="text/css" media="screen" href="../CSS/HeaderCSS.css" >
+    <script src = "../JavaScipt/HeaderJS.js"></script>
+    
+    <meta charset="UTF-8">
+    <title>Practica 3</title>
+    <meta name="description" content="Practica 3" />
+    
+</head>
+
+<body>
+    <header class="full" id="header">
+        <span id="x"></span>  
+        
+        <img id = "logo" src="../images/logo.png" width="120" height="40" alt="logo">
+        
+        <img id="lupa" src="../images/busqueda.svg" width="40" height="40" alt="lupa">
+        
+    </header>
+    <main>
+        <div id="menu2" >
+            <p id = "m1"></p>
+            <p id = "m2"></p>
+            <p id = "m3"></p>
+            <p id = "m4"></p>
+            <p id = "m5"></p>
+        </div>
+    </main>
 </body>
\ No newline at end of file
diff --git a/JavaScipt/HeaderJS.js b/JavaScipt/HeaderJS.js
index 568fc3e..68d9f8f 100644
--- a/JavaScipt/HeaderJS.js
+++ b/JavaScipt/HeaderJS.js
@@ -1,123 +1,124 @@
-var fichero;
-
-//Array con la información de todas las gafas del listado
-var array_gafas;
-
-const token = '5e18ba3a6b97a';
-
-
-fetch('http://puigpedros.salle.url.edu/pwi/glasses/api/list/' + token)
-.then(function(response) {
-    return response.json();
-    
-})
-.then(function(myJson) {
-    fichero = myJson;             
-});
-
-class Gafa {
-    constructor (data_gafa) {
-        this.id = data_gafa.id;
-        this.title = data_gafa.title;
-        this.description = data_gafa.description;
-        this.price = data_gafa.price;
-        this.images = data_gafa.images;
-        this.colors = data_gafa.colors;
-        this.details = data_gafa.details;
-        this.bgColor = data_gafa.bgColor;
-    }
-}
-
-window.onload = function() {
-    document.getElementById('x').addEventListener('click', function () {
-        if (this.classList.contains('clicked')) {
-          this.classList.remove('clicked');
-            document.getElementById("header").classList.add("full");
-            document.body.style.backgroundColor = "white";
-            document.body.style.opacity = 1;
-            
-            document.getElementById("menu2").style.opacity = "0";
-            document.getElementById("logo").style.opacity = "1";
-            document.getElementById("lupa").style.opacity = "1";
-        } else {
-            var n_gafas = pide();
-            document.getElementById("header").classList.remove("full");
-            //Reducción de la opacidad del fondo para simular efecto
-            document.body.style.opacity = 0.3;
-            document.getElementById("menu2").style.opacity = "1";
-            document.getElementById("logo").style.opacity = "0";
-            document.getElementById("lupa").style.opacity = "0";
-            actualizaMenu(n_gafas);
-
-            this.classList.add('clicked');
-        }
-      });
-
-}
-
-
-  function actualizaMenu (n_gafas) {
-    console.log("brilli brilli");
-    document.getElementById("m1").innerHTML = "HOME";
-    document.getElementById("m2").innerHTML = "LIST (" + n_gafas + ")";
-    document.getElementById("m3").innerHTML = "SEARCH";
-    console.log(document.getElementById("m1").innerHTML);
-    var n_cesta = recuperaCesta();
-    if (n_cesta == null) {
-        document.getElementById("m4").innerHTML = "CART (0)";
-    } else {
-        document.getElementById("m4").innerHTML = "CART (" + n_cesta + ")";
-    }
-          
-    document.getElementById("m5").innerHTML = "FAVORITE";  
-  }
-
-  
-
-function pide () {
-    
-    var url = 'http://puigpedros.salle.url.edu/pwi/glasses/api/list/' + token;
-    function get(url) {
-
-        fetch(url)
-            .then(function(response) {
-                return response.json();
-                
-            })
-            .then(function(myJson) {
-                fichero = myJson;             
-        });
-
-    };
-
-    get(url);
-    
-    array_gafas = new Array();
-    
-    for (var i = 0; i < fichero.data.glasses.length; i++) {
-        array_gafas[i] = new Gafa(fichero.data.glasses[i]);
-    }
-
-    return fichero.data.glasses.length;
-    
-    
-}
-
-
-function recuperaCesta () {
-    var cesta_json = localStorage.getItem('Cesta');
-    if (cesta_json == null) {
-        alert ("No hay ninguna cesta");
-    } else {
-        var cesta_obj = JSON.parse(cesta_json);
-
-        var numero = 0;
-        for (var i = 0; i < cesta_obj.length; i++) {
-            numero += cesta_obj [i][1];
-        }
-        return numero;
-    }
-    return null;
-    
-}
-
+var fichero;
+
+//Array con la información de todas las gafas del listado
+var array_gafas;
+
+const token = '5e18ba3a6b97a';
+
+
+fetch('http://puigpedros.salle.url.edu/pwi/glasses/api/list/' + token)
+.then(function(response) {
+    return response.json();
+    
+})
+.then(function(myJson) {
+    fichero = myJson;             
+});
+
+class Gafa {
+    constructor (data_gafa) {
+        this.id = data_gafa.id;
+        this.title = data_gafa.title;
+        this.description = data_gafa.description;
+        this.price = data_gafa.price;
+        this.images = data_gafa.images;
+        this.colors = data_gafa.colors;
+        this.details = data_gafa.details;
+        this.bgColor = data_gafa.bgColor;
+    }
+}
+/*
+window.onload = function() {
+
+    document.getElementById("x").addEventListener('click', function () {
+        if (this.classList.contains('clicked')) {
+          this.classList.remove('clicked');
+            document.getElementById("header").classList.add("full");
+            document.body.style.backgroundColor = "white";
+            document.body.style.opacity = 1;
+            
+            document.getElementById("menu2").style.opacity = "0";
+            document.getElementById("logo").style.opacity = "1";
+            document.getElementById("lupa").style.opacity = "1";
+        } else {
+            var n_gafas = pide();
+            document.getElementById("header").classList.remove("full");
+            //Reducción de la opacidad del fondo para simular efecto
+            document.body.style.opacity = 0.3;
+            document.getElementById("menu2").style.opacity = "1";
+            document.getElementById("logo").style.opacity = "0";
+            document.getElementById("lupa").style.opacity = "0";
+            actualizaMenu(n_gafas);
+
+            this.classList.add('clicked');
+        }
+      });
+
+}
+*/
+
+  function actualizaMenu (n_gafas) {
+    console.log("brilli brilli");
+    document.getElementById("m1").innerHTML = "HOME";
+    document.getElementById("m2").innerHTML = "LIST (" + n_gafas + ")";
+    document.getElementById("m3").innerHTML = "SEARCH";
+    console.log(document.getElementById("m1").innerHTML);
+    var n_cesta = recuperaCesta();
+    if (n_cesta == null) {
+        document.getElementById("m4").innerHTML = "CART (0)";
+    } else {
+        document.getElementById("m4").innerHTML = "CART (" + n_cesta + ")";
+    }
+          
+    document.getElementById("m5").innerHTML = "FAVORITE";  
+  }
+
+  
+
+function pide () {
+    
+    var url = 'http://puigpedros.salle.url.edu/pwi/glasses/api/list/' + token;
+    function get(url) {
+
+        fetch(url)
+            .then(function(response) {
+                return response.json();
+                
+            })
+            .then(function(myJson) {
+                fichero = myJson;             
+        });
+
+    };
+
+    get(url);
+    
+    array_gafas = new Array();
+    
+    for (var i = 0; i < fichero.data.glasses.length; i++) {
+        array_gafas[i] = new Gafa(fichero.data.glasses[i]);
+    }
+
+    return fichero.data.glasses.length;
+    
+    
+}
+
+
+function recuperaCesta () {
+    var cesta_json = localStorage.getItem('Cesta');
+    if (cesta_json == null) {
+        alert ("No hay ninguna cesta");
+    } else {
+        var cesta_obj = JSON.parse(cesta_json);
+
+        var numero = 0;
+        for (var i = 0; i < cesta_obj.length; i++) {
+            numero += cesta_obj [i][1];
+        }
+        return numero;
+    }
+    return null;
+    
+}
+
diff --git a/JavaScipt/ListJS.js b/JavaScipt/ListJS.js
index 76b4b4e..1ca362a 100644
--- a/JavaScipt/ListJS.js
+++ b/JavaScipt/ListJS.js
@@ -1,302 +1,303 @@
-class GafaDetalle{
-    constructor (id, title, description, price, images, colors, details, bgColor){
-        this.id = id;
-        this.title = title;
-        this.description = description;
-        this.price = price;
-        this.images = images;
-        this.colors = colors;
-        this.details = details;
-        this.bgColor = bgColor;
-    }
-
-    getId(){
-        return this.id;
-    }
-
-    getTitle(){
-        return this.title;
-    }
-
-    getDescription(){
-        return this.description;
-    }
-
-    getPrice(){
-        return this.price;
-    }
-
-    getImage(){
-        return this.images[0];
-    }
-
-    getColors(){
-        return this.colors;
-    }
-
-    getDetails(){
-        return this.details;
-    }
-
-    getBgColor(){
-        return this.bgColor;
-    }
-
-}
-
-function extraeListado(){
-    const token = '5e18ba3a6b97a';
-    const url_listado= 'http://puigpedros.salle.url.edu/pwi/glasses/api/list/' + token;
-    fetch(url_listado)
-        .then(function(response) {
-            return response.json();  
-        })
-        .then(function(listado) {
-            muestraListado(listado);
-    });
-}
-
-
-function muestraListado (listado) { 
-    //creamos un nodo que contendrá todas las gafas
-    var items = document.createElement("SECTION");
-    items.className = "items";
-    var cantidadGafas = listado.data.glasses.length;
-    for (var i = cantidadGafas - 1; i >= 0 ; i--) {
-        //gafa será el padre
-        var gafaNueva = document.createElement("ARTICLE");
-        gafaNueva.className = "item ";
-
-        //imagen, nombre y precio seran sus hijos
-        //imagen
-        var imagen = document.createElement("IMG");
-        imagen.src = listado.data.glasses[i].images[0];
-        imagen.className = "imagenGafas";
-        gafaNueva.appendChild(imagen);
-
-        //tabla para el nombre y el precio
-        var tabla = document.createElement("TABLE");
-        tabla.className = "tableItem";
-        var fila = tabla.insertRow();
-        fila.className = "tableRow";
-
-       //nombre
-        var primeraCelda = fila.insertCell();
-        primeraCelda.appendChild(document.createTextNode(listado.data.glasses[i].title));
-
-        //precio
-        var segundaCelda = fila.insertCell();
-        segundaCelda.className = "columnaItem";
-        segundaCelda.appendChild(document.createTextNode(listado.data.glasses[i].price + " €"));
-
-        gafaNueva.appendChild(tabla);
-
-        var link = document.createElement('a');
-        var linkText = document.createTextNode("Más información");
-        link.appendChild(linkText);
-        link.className = "link";
-        var id = listado.data.glasses[i].id;
-        function cambiaDetalle(id) {
-            return function() {extraeDetalle(id)};
-        }
-        link.addEventListener("click",cambiaDetalle(id));
-        gafaNueva.appendChild(link);
-        gafaNueva.style.backgroundColor = listado.data.glasses[i].bgColor;
-        items.appendChild(gafaNueva);
-    }
-    document.getElementById("detalle").innerHTML= "";
-    document.getElementById("detalle").style.display = "none";
-    document.getElementById("listado").appendChild(items);
-    document.getElementById("listado").style.display = "block";
-    var historial = {foo: "bar"};
-    history.pushState(historial, "", "#listado");
-}
-
-function extraeDetalle (id) { 
-    const token = '5dffb1e5acd7b';
-    var url_detalle= 'http://puigpedros.salle.url.edu/pwi/glasses/api/detail/' + token + '/' + id;
-    fetch(url_detalle)
-        .then(function(response) {
-            return response.json();  
-        })
-        .then(function(detalle) {
-            muestraDetalle(detalle);
-    });
-}
-
-function muestraDetalle (detalle) {
-    //creamos un nuevo item
-    var gafaDetalle = document.createElement("ARTICLE");
-
-    var gafa = new GafaDetalle(detalle.data.id, detalle.data.title, detalle.data.description, detalle.data.price, detalle.data.images, detalle.data.colors, detalle.data.details, detalle.data.bgColor);
-    gafaDetalle.className = "itemDetalle";
-    gafaDetalle.style.backgroundColor = gafa.getBgColor();
-    
-    //creamos la cabecera
-    var cabecera = document.createElement("ARTICLE");
-    cabecera.className = "cabecera";
-    //creamos el nombre de la gafa
-    var tituloPrincipal = document.createElement("DIV");
-    tituloPrincipal.appendChild(document.createTextNode(gafa.getTitle()));
-    tituloPrincipal.className = "tituloPrincipal";
-    cabecera.appendChild(tituloPrincipal);
-    //creamos el boton like
-    var like = document.createElement("object");
-    like.type="image/svg+xml";
-    like.src = "../images/corazon.svg";
-    like.className = "like";
-    cabecera.appendChild(like);
-    gafaDetalle.appendChild(cabecera);
-
-    //creamos la seccion de las gafas
-    var gafas = document.createElement("SECTION");
-    gafas.className = "gafas";
-    //creamos la imagen de las gafas
-    var imagenGafas = document.createElement("IMG");
-    imagenGafas.className = "imagenGafasDetalle";
-    imagenGafas.src = gafa.getImage();
-    gafas.appendChild(imagenGafas);
-    gafaDetalle.appendChild(gafas);
-
-    //creamos la tabla para el precio y versiones de la gafa
-    var tablaGafa = document.createElement("TABLE");
-    tablaGafa.className = "tableItemDetalle";
-
-    //fila 1 tabla gafa
-    var fila1TGafa = tablaGafa.insertRow();
-    fila1TGafa.className = "fila1t1";
-    //columna 1 fila 1
-    var col1fila1TGafa = fila1TGafa.insertCell();
-    col1fila1TGafa.appendChild(document.createTextNode("VERSIONS & COLORS"));
-    //columna 2 fila 1
-    var col2fila1TGafa = fila1TGafa.insertCell();
-    col2fila1TGafa.appendChild(document.createTextNode("PRICE"));
-
-    //fila 2 tabla gafa
-    var fila2TGafa = tablaGafa.insertRow();
-    fila2TGafa.className = "fila2t1";
-    //columna 1 fila 2
-    var col1fila2TGafa = fila2TGafa.insertCell();
-    var select = document.createElement("SELECT");
-    select.className = "select";
-    for(var i = 0; i < gafa.getColors().length; i++){
-        let option = document.createElement("OPTION");
-        option.className = "option";
-        option.text = gafa.getColors()[i];
-        select.add(option);
-    }
-    col1fila2TGafa.appendChild(select);
-
-    //columna 2 fila 2
-    var col2fila2TGafa = fila2TGafa.insertCell();
-    col2fila2TGafa.appendChild(document.createTextNode(gafa.getPrice() + " €"))
-    gafaDetalle.appendChild(tablaGafa);
-
-    //Section info
-    var info = document.createElement("SECTION");
-    info.className = "info";
-    //Section filaBotones
-    var filaBotones = document.createElement("SECTION");
-    filaBotones.className = "filaBotones";
-    //Section boton
-    var botones = document.createElement("SECTION");
-    botones.className = "botones";
-    //Button BUY
-    //Creamos la cesta
-    var cesta = JSON.parse(localStorage.getItem('cesta'));
-    if(cesta === null){
-        cesta = new Array();
-    }
-    function añadirCesta(id) {
-        cesta.push(id);
-        //añadimos la cesta al localStorage
-        localStorage.clear();
-        localStorage.setItem('cesta', JSON.stringify(cesta));
-    }
-    
-    var buttonBuy = document.createElement('BUTTON');
-    var id = gafa.getId();
-    buttonBuy.addEventListener("click", function(){añadirCesta(id)});
-    buttonBuy.className = "boton";
-    buttonBuy.appendChild(document.createTextNode("BUY"));
-    botones.appendChild(buttonBuy);
-    //Button ATRÁS
-    function cambiaListado() {
-        return function() {extraeListado()};
-    }
-    var buttonBack = document.createElement('BUTTON');
-    buttonBack.className = "boton";
-    buttonBack.appendChild(document.createTextNode("BACK"));
-    buttonBack.addEventListener("click", cambiaListado());
-    botones.appendChild(buttonBack);
-    filaBotones.appendChild(botones);
-    info.appendChild(filaBotones);
-
-    //DIV Details
-    var details = document.createElement("DIV");
-    details.className = "titulo";
-    details.appendChild(document.createTextNode("DETAILS"));
-    info.appendChild(details);
-    
-    //Tabla para los details
-    var tablaDetails = document.createElement("TABLE");
-    tablaDetails.className = "tablaDetails";
-
-    //fila 1 tabla details
-    var fila1TDetails = tablaDetails.insertRow();
-    fila1TDetails.className = "fila1t2";
-    //columna 1 fila 1
-    var col1fila1TD = fila1TDetails.insertCell();
-    col1fila1TD.appendChild(document.createTextNode(gafa.getDetails().lensesDiameter));
-    col1fila1TD.className = "celda";
-
-    //columna 2 fila 1
-    var col2fila1TD = fila1TDetails.insertCell();
-    col2fila1TD.appendChild(document.createTextNode(gafa.getDetails().bridgeMaterial));
-    col2fila1TD.className = "celda";
-
-    //columna 3 fila 1
-    var col3fila1TD = fila1TDetails.insertCell();
-    col3fila1TD.appendChild(document.createTextNode(gafa.getDetails().rimsMaterial));
-    col3fila1TD.className = "celda";
-
-    //fila 2 tabla details
-    var fila2TDetails = tablaDetails.insertRow();
-    fila2TDetails.className = "fila2t2";
-    //columna 1 fila 2
-    var col1fila2TD = fila2TDetails.insertCell();
-    col1fila2TD.appendChild(document.createTextNode("Lenses diameter"));
-    col1fila2TD.className = "celda";
-
-    //columna 2 fila 2
-    var col2fila2TD = fila2TDetails.insertCell();
-    col2fila2TD.appendChild(document.createTextNode("Bridge material"));
-    col2fila2TD.className = "celda";
-
-    //columna 3 fila 2
-    var col3fila2TD = fila2TDetails.insertCell();
-    col3fila2TD.appendChild(document.createTextNode("Rims material"));
-    col3fila2TD.className = "celda";
-    info.appendChild(tablaDetails);
-    
-    //DIV DESCRIPTION
-    var description = document.createElement("DIV");
-    description.className = "titulo";
-    description.appendChild(document.createTextNode("DESCRIPTION"));
-    info.appendChild(description);
-
-    //Description text
-    var descriptionText = document.createElement("P");
-    descriptionText.className = "texto";
-    descriptionText.appendChild(document.createTextNode(gafa.getDescription()));
-    info.appendChild(descriptionText);
-
-    gafaDetalle.appendChild(info);
-
-    document.getElementById("listado").innerHTML= "";
-    document.getElementById("listado").style.display = "none";
-    document.getElementById("detalle").appendChild(gafaDetalle);
-    document.getElementById("detalle").style.display = "block";
-    var historial = {foo: "bar"};
-    history.pushState(historial, "", "#detalle");
-}
+class GafaDetalle{
+    constructor (id, title, description, price, images, colors, details, bgColor){
+        this.id = id;
+        this.title = title;
+        this.description = description;
+        this.price = price;
+        this.images = images;
+        this.colors = colors;
+        this.details = details;
+        this.bgColor = bgColor;
+    }
+
+    getId(){
+        return this.id;
+    }
+
+    getTitle(){
+        return this.title;
+    }
+
+    getDescription(){
+        return this.description;
+    }
+
+    getPrice(){
+        return this.price;
+    }
+
+    getImage(){
+        return this.images[0];
+    }
+
+    getColors(){
+        return this.colors;
+    }
+
+    getDetails(){
+        return this.details;
+    }
+
+    getBgColor(){
+        return this.bgColor;
+    }
+
+}
+
+function extraeListado(){
+    const token = '5e18ba3a6b97a';
+    const url_listado= 'http://puigpedros.salle.url.edu/pwi/glasses/api/list/' + token;
+    fetch(url_listado)
+        .then(function(response) {
+            return response.json();  
+        })
+        .then(function(listado) {
+            muestraListado(listado);
+    });
+}
+
+
+function muestraListado (listado) { 
+    //creamos un nodo que contendrá todas las gafas
+    var items = document.createElement("SECTION");
+    items.className = "items";
+    var cantidadGafas = listado.data.glasses.length;
+    for (var i = cantidadGafas - 1; i >= 0 ; i--) {
+        //gafa será el padre
+        var gafaNueva = document.createElement("ARTICLE");
+        gafaNueva.className = "item ";
+
+        //imagen, nombre y precio seran sus hijos
+        //imagen
+        var imagen = document.createElement("IMG");
+        imagen.src = listado.data.glasses[i].images[0];
+        imagen.className = "imagenGafas";
+        gafaNueva.appendChild(imagen);
+
+        //tabla para el nombre y el precio
+        var tabla = document.createElement("TABLE");
+        tabla.className = "tableItem";
+        var fila = tabla.insertRow();
+        fila.className = "tableRow";
+
+       //nombre
+        var primeraCelda = fila.insertCell();
+        primeraCelda.appendChild(document.createTextNode(listado.data.glasses[i].title));
+
+        //precio
+        var segundaCelda = fila.insertCell();
+        segundaCelda.className = "columnaItem";
+        segundaCelda.appendChild(document.createTextNode(listado.data.glasses[i].price + " €"));
+
+        gafaNueva.appendChild(tabla);
+
+        var link = document.createElement('a');
+        var linkText = document.createTextNode("Más información");
+        link.appendChild(linkText);
+        link.className = "link";
+        link.id = "link";
+        var id = listado.data.glasses[i].id;
+        function cambiaDetalle(id) {
+            return function() {extraeDetalle(id)};
+        }
+        link.addEventListener("click",cambiaDetalle(id));
+        gafaNueva.appendChild(link);
+        gafaNueva.style.backgroundColor = listado.data.glasses[i].bgColor;
+        items.appendChild(gafaNueva);
+    }
+    document.getElementById("detalle").innerHTML= "";
+    document.getElementById("detalle").style.display = "none";
+    document.getElementById("listado").appendChild(items);
+    document.getElementById("listado").style.display = "block";
+    var historial = {foo: "bar"};
+    history.pushState(historial, "", "#listado");
+}
+
+function extraeDetalle (id) { 
+    const token = '5dffb1e5acd7b';
+    var url_detalle= 'http://puigpedros.salle.url.edu/pwi/glasses/api/detail/' + token + '/' + id;
+    fetch(url_detalle)
+        .then(function(response) {
+            return response.json();  
+        })
+        .then(function(detalle) {
+            muestraDetalle(detalle);
+    });
+}
+
+function muestraDetalle (detalle) {
+    //creamos un nuevo item
+    var gafaDetalle = document.createElement("ARTICLE");
+
+    var gafa = new GafaDetalle(detalle.data.id, detalle.data.title, detalle.data.description, detalle.data.price, detalle.data.images, detalle.data.colors, detalle.data.details, detalle.data.bgColor);
+    gafaDetalle.className = "itemDetalle";
+    gafaDetalle.style.backgroundColor = gafa.getBgColor();
+    
+    //creamos la cabecera
+    var cabecera = document.createElement("ARTICLE");
+    cabecera.className = "cabecera";
+    //creamos el nombre de la gafa
+    var tituloPrincipal = document.createElement("DIV");
+    tituloPrincipal.appendChild(document.createTextNode(gafa.getTitle()));
+    tituloPrincipal.className = "tituloPrincipal";
+    cabecera.appendChild(tituloPrincipal);
+    //creamos el boton like
+    var like = document.createElement("object");
+    like.type="image/svg+xml";
+    like.src = "../images/corazon.svg";
+    like.className = "like";
+    cabecera.appendChild(like);
+    gafaDetalle.appendChild(cabecera);
+
+    //creamos la seccion de las gafas
+    var gafas = document.createElement("SECTION");
+    gafas.className = "gafas";
+    //creamos la imagen de las gafas
+    var imagenGafas = document.createElement("IMG");
+    imagenGafas.className = "imagenGafasDetalle";
+    imagenGafas.src = gafa.getImage();
+    gafas.appendChild(imagenGafas);
+    gafaDetalle.appendChild(gafas);
+
+    //creamos la tabla para el precio y versiones de la gafa
+    var tablaGafa = document.createElement("TABLE");
+    tablaGafa.className = "tableItemDetalle";
+
+    //fila 1 tabla gafa
+    var fila1TGafa = tablaGafa.insertRow();
+    fila1TGafa.className = "fila1t1";
+    //columna 1 fila 1
+    var col1fila1TGafa = fila1TGafa.insertCell();
+    col1fila1TGafa.appendChild(document.createTextNode("VERSIONS & COLORS"));
+    //columna 2 fila 1
+    var col2fila1TGafa = fila1TGafa.insertCell();
+    col2fila1TGafa.appendChild(document.createTextNode("PRICE"));
+
+    //fila 2 tabla gafa
+    var fila2TGafa = tablaGafa.insertRow();
+    fila2TGafa.className = "fila2t1";
+    //columna 1 fila 2
+    var col1fila2TGafa = fila2TGafa.insertCell();
+    var select = document.createElement("SELECT");
+    select.className = "select";
+    for(var i = 0; i < gafa.getColors().length; i++){
+        let option = document.createElement("OPTION");
+        option.className = "option";
+        option.text = gafa.getColors()[i];
+        select.add(option);
+    }
+    col1fila2TGafa.appendChild(select);
+
+    //columna 2 fila 2
+    var col2fila2TGafa = fila2TGafa.insertCell();
+    col2fila2TGafa.appendChild(document.createTextNode(gafa.getPrice() + " €"))
+    gafaDetalle.appendChild(tablaGafa);
+
+    //Section info
+    var info = document.createElement("SECTION");
+    info.className = "info";
+    //Section filaBotones
+    var filaBotones = document.createElement("SECTION");
+    filaBotones.className = "filaBotones";
+    //Section boton
+    var botones = document.createElement("SECTION");
+    botones.className = "botones";
+    //Button BUY
+    //Creamos la cesta
+    var cesta = JSON.parse(localStorage.getItem('cesta'));
+    if(cesta === null){
+        cesta = new Array();
+    }
+    function añadirCesta(id) {
+        cesta.push(id);
+        //añadimos la cesta al localStorage
+        localStorage.clear();
+        localStorage.setItem('cesta', JSON.stringify(cesta));
+    }
+    
+    var buttonBuy = document.createElement('BUTTON');
+    var id = gafa.getId();
+    buttonBuy.addEventListener("click", function(){añadirCesta(id)});
+    buttonBuy.className = "boton";
+    buttonBuy.appendChild(document.createTextNode("BUY"));
+    botones.appendChild(buttonBuy);
+    //Button ATRÁS
+    function cambiaListado() {
+        return function() {extraeListado()};
+    }
+    var buttonBack = document.createElement('BUTTON');
+    buttonBack.className = "boton";
+    buttonBack.appendChild(document.createTextNode("BACK"));
+    buttonBack.addEventListener("click", cambiaListado());
+    botones.appendChild(buttonBack);
+    filaBotones.appendChild(botones);
+    info.appendChild(filaBotones);
+
+    //DIV Details
+    var details = document.createElement("DIV");
+    details.className = "titulo";
+    details.appendChild(document.createTextNode("DETAILS"));
+    info.appendChild(details);
+    
+    //Tabla para los details
+    var tablaDetails = document.createElement("TABLE");
+    tablaDetails.className = "tablaDetails";
+
+    //fila 1 tabla details
+    var fila1TDetails = tablaDetails.insertRow();
+    fila1TDetails.className = "fila1t2";
+    //columna 1 fila 1
+    var col1fila1TD = fila1TDetails.insertCell();
+    col1fila1TD.appendChild(document.createTextNode(gafa.getDetails().lensesDiameter));
+    col1fila1TD.className = "celda";
+
+    //columna 2 fila 1
+    var col2fila1TD = fila1TDetails.insertCell();
+    col2fila1TD.appendChild(document.createTextNode(gafa.getDetails().bridgeMaterial));
+    col2fila1TD.className = "celda";
+
+    //columna 3 fila 1
+    var col3fila1TD = fila1TDetails.insertCell();
+    col3fila1TD.appendChild(document.createTextNode(gafa.getDetails().rimsMaterial));
+    col3fila1TD.className = "celda";
+
+    //fila 2 tabla details
+    var fila2TDetails = tablaDetails.insertRow();
+    fila2TDetails.className = "fila2t2";
+    //columna 1 fila 2
+    var col1fila2TD = fila2TDetails.insertCell();
+    col1fila2TD.appendChild(document.createTextNode("Lenses diameter"));
+    col1fila2TD.className = "celda";
+
+    //columna 2 fila 2
+    var col2fila2TD = fila2TDetails.insertCell();
+    col2fila2TD.appendChild(document.createTextNode("Bridge material"));
+    col2fila2TD.className = "celda";
+
+    //columna 3 fila 2
+    var col3fila2TD = fila2TDetails.insertCell();
+    col3fila2TD.appendChild(document.createTextNode("Rims material"));
+    col3fila2TD.className = "celda";
+    info.appendChild(tablaDetails);
+    
+    //DIV DESCRIPTION
+    var description = document.createElement("DIV");
+    description.className = "titulo";
+    description.appendChild(document.createTextNode("DESCRIPTION"));
+    info.appendChild(description);
+
+    //Description text
+    var descriptionText = document.createElement("P");
+    descriptionText.className = "texto";
+    descriptionText.appendChild(document.createTextNode(gafa.getDescription()));
+    info.appendChild(descriptionText);
+
+    gafaDetalle.appendChild(info);
+
+    document.getElementById("listado").innerHTML= "";
+    document.getElementById("listado").style.display = "none";
+    document.getElementById("detalle").appendChild(gafaDetalle);
+    document.getElementById("detalle").style.display = "block";
+    var historial = {foo: "bar"};
+    history.pushState(historial, "", "#detalle");
+}
diff --git a/index.html b/index.html
index e8d5b63..d85017e 100644
--- a/index.html
+++ b/index.html
@@ -1,64 +1,79 @@
-<!DOCTYPE html>
-<html lang="es">
-<head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <meta name="keywords" content="index, página principal">
-    <meta name="author" content="Bernardo Nouel, Sandra Cea, Rogelio Sansaloni">
-    <meta http-equiv="X-UA-Compatible" content="ie=edge">
-    <title>Practica 4</title>
-    <style>
-        body.initialBody{
-            display:grid;
-            height: 100vh;
-            grid-template-columns: 100%;
-            grid-template-rows: 7% 78% 10%;
-            overflow-y: hidden;
-        }
-
-        body.bigHeader{
-            display:grid;
-            height: 100vh;
-            grid-template-columns: 100%;
-            grid-template-rows: 85% 0 10%;
-            overflow-y: hidden;
-        }
-
-        #header {
-            overflow: hidden;
-            padding-top: 10px;
-        }
-
-        #list {
-            height: 100%;
-        }
-
-    </style>
-</head>
-<body class="initialBody">
-    <iframe id="header" src="./HTML/Header.html" width="100%" height="100%" frameborder="0"></iframe>
-
-    <iframe id="list" src="./HTML/List.html" width="100%" frameborder="0"></iframe>
-
-    <iframe id="footer" src="./HTML/Footer.html" scrolling="no" width="100%" height="100vh" frameborder="0"></iframe>
-
-    <script>
-        document.getElementById("header").addEventListener('load',function(e){
-            let doc = document.getElementById("header").contentWindow.document;
-            doc.getElementById("x").addEventListener("click", () => {
-                console.log("holi");
-
-                if(document.body.classList.contains("initialBody")){
-                    document.body.classList.remove("initialBody");
-                    document.body.classList.add("bigHeader");
-                    document.getElementById("header").height = "100%";
-                }else{
-                    document.body.classList.remove("bigHeader");
-                    document.body.classList.add("initialBody");
-                    document.getElementById("header").height = "40vh";
-                }
-            })
-        });
-    </script>
-</body>
+<!DOCTYPE html>
+<html lang="es">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta name="keywords" content="index, página principal">
+    <meta name="author" content="Bernardo Nouel, Sandra Cea, Rogelio Sansaloni">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Practica 4</title>
+    <style>
+        body.initialBody{
+            display:grid;
+            height: 100vh;
+            grid-template-columns: 100%;
+            grid-template-rows: 7% 78% 10%;
+            overflow-y: hidden;
+        }
+
+        body.bigHeader{
+            display:grid;
+            height: 100vh;
+            grid-template-columns: 100%;
+            grid-template-rows: 85% 0 10%;
+            overflow-y: hidden;
+        }
+
+        #header {
+            overflow: hidden;
+            padding-top: 10px;
+        }
+
+        #list {
+            height: 100%;
+        }
+
+
+
+    </style>
+</head>
+<body class="initialBody">
+    <iframe id="header" src="./HTML/Header.html" width="100%" height="100%" frameborder="0"></iframe>
+
+    <iframe id="list" src="./HTML/List.html" width="100%" frameborder="0"></iframe>
+
+    <iframe id="footer" src="./HTML/Footer.html" scrolling="no" width="100%" height="100vh" frameborder="0"></iframe>
+
+    <script>
+        document.getElementById("header").addEventListener('load',function(e){
+            let doc = document.getElementById("header").contentWindow.document;
+            doc.getElementById("x").addEventListener("click", () => {
+                console.log("holi");
+
+                if(document.body.classList.contains("initialBody")){
+                    document.body.classList.remove("initialBody");
+                    document.body.classList.add("bigHeader");
+                    document.getElementById("header").height = "100%";
+                }else{
+                    document.body.classList.remove("bigHeader");
+                    document.body.classList.add("initialBody");
+                    document.getElementById("header").height = "40vh";
+                }
+            })
+        });
+
+        document.getElementById("list").addEventListener('load', function(e){
+            console.log("a");
+            let linkDoc = document.getElementById("list").contentWindow.document;
+            console.log(linkDoc);
+            linkDoc.
+            linkDoc.getElementById("link").addEventListener("click", () => {
+                console.log("eh");
+            });
+
+        });
+
+
+    </script>
+</body>
 </html>
\ No newline at end of file
